src/pages/ProfilePage/ProfilePage.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import UserCard from '../../components/UserCard';
+import UserPostList from '../../components/UserPostList';
+import ErrorAlert, {
+  renderErrorAlert,
+} from '../../components/ErrorAlert/ErrorAlert';
+import { ErrorMessages } from '../../constants';
+import { UserInfo } from '../../types/userInfoTypes';
+import { ErrorObject } from '../../types/trendingFeedTypes';
+import { fetchData } from '../../api';
+import { Endpoint, currentUser } from '../../api/constants';
+
+const ProfilePage = () => {
+  const params = useParams();
+  const chosenUser = params.uniqueId;
+  const user = chosenUser || currentUser;
+
+  const [profile, setProfile] = useState({});
+  const [userPosts, setUserPosts] = useState([]);
+
+  const [postsIsLoading, setPostsIsLoading] = useState(false);
+  const [profileIsLoading, setProfileIsLoading] = useState(false);
+
+  const [postsIsError, setPostsIsError] = useState(false);
+  const [profileIsError, setProfileIsError] = useState(false);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    fetchData(
+      Endpoint.UserFeed + user,
+      controller,
+      setPostsIsLoading,
+      setUserPosts,
+      setPostsIsError
+    );
+    return () => controller.abort();
+  }, [user]);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    fetchData(
+      Endpoint.UserInfo + user,
+      controller,
+      setProfileIsLoading,
+      setProfile,
+      setProfileIsError
+    );
+    return () => controller.abort();
+  }, [user]);
+
+  if (
+    Object.keys(profile).includes('message') ||
+    postsIsError ||
+    profileIsError
```

Comment:
In the case of conditional rendering with more than 2 dependencies, probably better to merge all dependencies into one variable and use it